data/lib/api/ball_score/ball_score_model.dart

```diff
@@ -827,23 +759,18 @@ extension BallScoreList on List<BallScoreModel> {
     final bowledBallCount = deliveries
         .where(
           (element) =>
+              element.extras_type != ExtrasType.penaltyRun &&
               element.wicket_type != WicketType.retired &&
               element.wicket_type != WicketType.retiredHurt &&
-              element.wicket_type != WicketType.timedOut &&
-              element.extras_type != ExtrasType.penaltyRun,
+              element.wicket_type != WicketType.timedOut,
         )
         .length;
 
     final bowledBallCountForEconomyRate = deliveries
         .where(
-          (element) =>
-              (element.extras_type == null ||
-                  element.extras_type == ExtrasType.legBye ||
-                  element.extras_type == ExtrasType.bye) &&
-              element.wicket_type != WicketType.retired &&
-              element.wicket_type != WicketType.retiredHurt &&
-              element.wicket_type != WicketType.timedOut &&
-              element.extras_type != ExtrasType.penaltyRun,
+          (element) => (element.extras_type == null ||
+              element.extras_type == ExtrasType.legBye ||
+              element.extras_type == ExtrasType.bye),
```

Comment:
economy will be count for actually thrown ball, previous condition where we checked for wicket type is correct because this four dismissals are there for records but do not count as ball delivery, let's revert this change.